File: app/src/main/java/ahmadadam96/comicbookmovies/Movie.java
package ahmadadam96.comicbookmovies;


import android.os.Parcel;
import android.os.Parcelable;

import java.text.ParsePosition;
import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by ahmad on 2017-03-14.
 */

public class Movie implements Parcelable {
    //The release date of the movie
    private Date mReleaseDate;

    //The title of the movie
    private String mTitle;

    //A short overview talking about what the movie is about
    private String mOverview;

    //The official poster for the movie
    private String mPosterUrl;

    //The URL to the official movie page (not always available
    private String mUrl;

    //The IMDB id to link to the IMDB page
    private String mIMDBId;

    //Universe the movie is set in
    private String mUniverse;

    //The constructor for the Movie class with a universe
    public Movie(String releaseDate, String title, String overview,
                 String posterUrl, String url, String IMDBId, String universe) {
        mReleaseDate = convertDate(releaseDate);
        mTitle = title;
        mOverview = overview;
        mPosterUrl = posterUrl;
        mUrl = url;
        mIMDBId = IMDBId;
        mUniverse = universe;
    }

    //The constructor for the Movie class without a universe
    public Movie(String releaseDate, String title, String overview,
                 String posterUrl, String url, String IMDBId) {
        mReleaseDate = convertDate(releaseDate);
        mTitle = title;
        mOverview = overview;
        mPosterUrl = posterUrl;
        mUrl = url;
        mIMDBId = IMDBId;
        mUniverse = "Unknown";
    }

    public Movie(Parcel in) {
        mTitle = in.readString();
        mOverview = in.readString();
        mPosterUrl = in.readString();
        mUrl = in.readString();
        mIMDBId = in.readString();
        mUniverse = in.readString();
        mReleaseDate = (Date) in.readSerializable();
    }

    //A function to convert the date from a string to a dateObject
    // with the correct format
    private Date convertDate(String releaseDate) {
        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
        return dateFormat.parse(releaseDate, new ParsePosition(0));
    }

    //Member get methods for each object
    public Date getReleaseDate() {
        return mReleaseDate;
    }

    public String getTitle() {
        if (mTitle != null) {
            return mTitle;
        } else return "Unknown";
    }

    public String getOverview() {
        if (mOverview != null) {
            return mOverview;
        } else return "Unknown";
    }

    public String getPosterUrl() {
        if (mPosterUrl != null) {
            return mPosterUrl;
        } else return "Unknown";
    }

    public String getUrl() {
        if (mUrl != null) {
            return mUrl;
        } else return "Unknown";
    }

    public String getIMDBId() {
        if (mIMDBId != null) {
            return mIMDBId;
        } else return "Unknown";
    }

    public String getUniverse() {
        if (mUniverse != null) {
            return mUniverse;
        } else return "Unknown";
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(mTitle);
        dest.writeString(mOverview);
        dest.writeString(mPosterUrl);
        dest.writeString(mUrl);
        dest.writeString(mIMDBId);
        dest.writeString(mUniverse);
        dest.writeSerializable(mReleaseDate);
    }

    //Creator
    public static final Parcelable.Creator CREATOR = new Parcelable.Creator() {
        @Override
        public Movie createFromParcel(Parcel source) {
            return new Movie(source);
        }

        @Override
        public Movie[] newArray(int size) {
            return new Movie[size];
        }
    };
}

File: app/src/main/java/ahmadadam96/comicbookmovies/Movie.kt
package ahmadadam96.comicbookmovies

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.text.ParsePosition
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by Ahmed El Yaacoub on 2017-03-14.
 */

@Parcelize
class Movie() : Parcelable {

    //The release date of the movie
    //Member get methods for each object
    var releaseDate: Date = Calendar.getInstance().time

    //The title of the movie
    var title: String = "Unknown"

    //A short overview talking about what the movie is about
    var overview: String = "Unknown"

    //The official poster for the movie
    var posterUrl: String = "Unknown"

    //The URL to the official movie page (not always available
    var url: String = "Unknown"

    //The IMDB id to link to the IMDB page
    var IMDBId: String = "Unknown"

    //Universe the movie is set in
    var universe: String = "Unknown"

    constructor(releaseDate: String, title: String, overview: String,
                posterUrl: String, url: String, IMDBId: String, universe: String) : this() {
        this.releaseDate = convertDate(releaseDate)
        this.title = title
        this.overview = overview
        this.posterUrl = posterUrl
        this.url = url
        this.IMDBId = IMDBId
        this.universe = universe
    }

    //The constructor for the Movie class without a universe
    constructor(releaseDate: String, title: String, overview: String,
                posterUrl: String, url: String, IMDBId: String) : this() {
        this.releaseDate = convertDate(releaseDate)
        this.title = title
        this.overview = overview
        this.posterUrl = posterUrl
        this.url = url
        this.IMDBId = IMDBId
        this.universe = "Unknown"
    }


    //A function to convert the date from a string to a dateObject
    // with the correct format
    private fun convertDate(releaseDate: String): Date {
        val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.getDefault())
        return dateFormat.parse(releaseDate, ParsePosition(0))
    }
}
